### PR TITLE
LPP-51452 Add logs

### PR DESCRIPTION
**Pull request by gustavolimav**
https://liferay.atlassian.net/browse/LPP-51964

### CODE REVIEW

**Comment by gustavolimav** [comment]
Hi @joshuacords,

Today I added log messages to the functions responsible for updating the Elasticsearch settings, to the classes that manage the connection to Elasticsearch and to the class responsible for displaying the connection information. Could you review this initial PR when you get a chance? Your feedback would be very helpful. 

I'll take a look at it again tomorrow. I only had time to add these logs today. Thanks!

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/configuration/ElasticsearchConfigurationWrapper.java [hunk 1]
```diff
@@ -283,9 +285,17 @@ protected void activate(Map<String, Object> map) {
 			ElasticsearchConfiguration.class, propsMap);
 		_propsMap = propsMap;
 
+		if (_log.isInfoEnabled()) {
+			_log.info("Start Update Elasticsearch Configuration");
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/configuration/ElasticsearchConfigurationWrapper.java:289]: Let's log what the configuration is.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManager.java [hunk 2]
```diff
@@ -55,6 +55,10 @@ public class ElasticsearchConnectionManager
 	public void addElasticsearchConnection(
 		ElasticsearchConnection elasticsearchConnection) {
 
+		if (_log.isInfoEnabled()) {
+			_log.info("Start Adding Elasticsearch Connection");
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManager.java:59]: Log what the configuration is.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManager.java [hunk 3]
```diff
@@ -275,6 +283,12 @@ public void onElasticsearchConfigurationUpdate() {
...
 	public void removeElasticsearchConnection(String connectionId) {
+		if (_log.isInfoEnabled()) {
+			_log.info(
+				"Start Removing Elasticsearch Connection with connection ID: " +
+					connectionId);
+		}
+
 		if (connectionId == null) {
 			return;
 		}
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManager.java:294]: Could we log what the _elasticsearchConnectionSupplier returns and if it's null.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManager.java [hunk 4]
```diff
@@ -306,12 +326,26 @@ protected void applyConfigurations() {
 			elasticsearchConfigurationWrapper.restClientLoggerLevel());
 
 		if (operationModeResolver.isProductionModeEnabled()) {
-			if (Validator.isBlank(
-					elasticsearchConfigurationWrapper.
-						remoteClusterConnectionId())) {
+			String remoteClusterConnectionId =
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchConnectionManager.java:329]: Could we log what the remoteClusterConnectionId is.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java [hunk 5]
```diff
@@ -219,6 +223,12 @@ private void _addConnectionInformation(
 		List<ConnectionInformation> connectionInformationList,
 		String... labels) {
 
+		if (_log.isInfoEnabled()) {
+			_log.info(
+				"Adding Connection Information to connection with ID: " +
+					elasticsearchConnection.getConnectionId());
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java:229]: Log everything in connection id, especially the name.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java [hunk 6]
```diff
@@ -219,6 +223,12 @@ private void _addConnectionInformation(
...
+		}
+
 		if (elasticsearchConnection == null) {
 			return;
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java:233]: Log if null

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java [hunk 7]
```diff
@@ -261,6 +271,12 @@ private void _addConnectionInformation(
 		}
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java:271]: What are these labels? If they are related to the connection names we should log them.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java [hunk 8]
```diff
@@ -190,6 +190,10 @@ private void _addActiveConnections(
 			filterString);
```
Comment (joshuacords) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/information/ElasticsearchSearchEngineInformation.java:190]: Where ever connectionInformationList is being passed from, we should have logging on that.

**Comment by joshuacords** [comment]
Good start Gustavo. As a mindset, we don't care how busy a debug patch logs are, we just want to make sure we capture anything that could be important that way we only need the customer to use it once. (That said, it's likely some of what I requested is redundant, but it's hard to tell without testing it.)

As you add to this, a comment with what output looks like could be helpful.

Thank you

**Review by joshuacords — CHANGES_REQUESTED** [review]
Take a look at these suggestions.

**Pull request closed without merging**